src/Components/MovieTile/__tests__/MovieTile.test.js

```diff
@@ -0,0 +1,51 @@
+import { render, screen } from '@testing-library/react';
+import { MovieTile } from '../MovieTile';
+import userEvent from '@testing-library/user-event';
+
+describe('MovieTile', () => {
+  test('renders all details', () => {
+    const testData = {
+      imageUrl: 'imageUrl',
+      movieName: 'testName',
+      releaseYear: 'testYear',
+      relevantGenres: ['genre1', 'genre2'],
+    };
+
+    render(<MovieTile {...testData} />);
+
+    expect(screen.getByRole('img')).toHaveAttribute('src', testData.imageUrl);
+    expect(screen.getByText(testData.movieName)).toBeInTheDocument();
+    expect(screen.getByText(testData.releaseYear)).toBeInTheDocument();
+    expect(screen.getByText('genre1 & genre2')).toBeInTheDocument();
+  });
+
+  test('renders without throwing an exception with undefined relevant genres', () => {
+    const testData = {
+      imageUrl: 'imageUrl',
+      movieName: 'testName',
+      releaseYear: 'testYear',
+      relevantGenres: undefined,
+    };
+
+    expect(() => {
+      render(<MovieTile {...testData} />);
+    }).not.toThrow();
+  });
```

Comment:
A test like this is not needed usually. 
If you want to check this case, there is an approach where you create 2 tests for components with optional props: "renders with all props" and "renders with required props". And you create snapshots in each test. `not.toThrow()` is not needed.